packages/devextreme/testing/tests/DevExpress.ui.widgets.form/form.tests.js

```diff
@@ -4540,6 +4541,98 @@ QUnit.test('form should be dirty when some editors are dirty', function(assert)
     assert.strictEqual(form.option('isDirty'), false, 'form is not dirty when all editors are back to pristine');
 });
 
+[true, false].forEach((openOnFieldClick) => {
+    [true, false].forEach((hideOnOutsideClickValue) => {
+        QUnit.test(`Opened DropDownList must hide on input label click, openOnFieldClick: ${openOnFieldClick}, hideOnOutsideClick: ${hideOnOutsideClickValue} (T1257945)`, function(assert) {
+            let initialHideOnOutsideClick;
+            const $form = $('#form').dxForm({
+                formData: { CustomerID: 'VINET' },
+                items: [{
+                    itemType: 'group',
+                    colCount: 2,
+                    items: [{
+                        dataField: 'CustomerID',
+                        editorType: 'dxSelectBox',
+                        editorOptions: {
+                            items: ['VINET', 'VALUE', 'VINS'],
+                            value: '',
+                            openOnFieldClick,
+                            dropDownOptions: {
+                                hideOnOutsideClick: () => hideOnOutsideClickValue,
+                                onContentReady: ({ component }) => {
+                                    initialHideOnOutsideClick = component.option('hideOnOutsideClick');
+                                }
+                            }
+                        },
+                    }],
+                }],
+            });
+
+            const $dropDownButton = $form.find('.dx-dropdowneditor-button');
+            $dropDownButton.trigger('click');
+
+            const hideOnOutsideClickSpy = sinon.spy(initialHideOnOutsideClick);
+            const formInstance = $form.dxForm('instance');
+            const editorInstance = formInstance.getEditor('CustomerID');
+
+            formInstance.option('items[0].items[0].editorOptions.dropDownOptions.hideOnOutsideClick', hideOnOutsideClickSpy);
+
+            assert.true(editorInstance.option('opened'), 'drop down list is visible');
+
+            const $label = $form.find('.dx-field-item-label-text');
+            pointerMock($label).click();
```

Comment:
Hm, does finally this click using `pointerMock` emulates real `label` behavior? I mean, doesn't it close the popup?
If it does, it's much better to omit hideOutsideClick spying and assert popup visibility state explicitly

Reply:
Unfortunately, it is not